ring/lifecycler.go

```diff
@@ -759,11 +779,42 @@ func (i *Lifecycler) compareTokens(fromRing Tokens) bool {
 	return true
 }
 
+func (i *Lifecycler) waitBeforeJoining(ctx context.Context) error {
+	retries := backoff.New(ctx, backoff.Config{
```

Comment:
Since I suggested to set MaxRetries = 0 you should wrap context with a `context.WithTimeout()`.